op-supervisor/supervisor/backend/db/logs/db.go

```diff
@@ -193,8 +199,10 @@ func (db *DB) OpenBlock(blockNum uint64) (ref eth.BlockRef, logCount uint32, exe
 	db.rwLock.RLock()
 	defer db.rwLock.RUnlock()
 
+	// TODO: need to fix assumption that the first block is block 0
+	// This is only true for genesis-Interop chains.
 	if blockNum == 0 {
-		seal, err := db.StartingBlock()
+		seal, err := db.FirstSealedBlock()
```

Comment:
`newIteratorAt` below handles the starting block case for non-genesis case. This `== 0` check is just to prevent the underflow.

Reply:
Fixed. And using the `types.ErrSkipped` error, which is meant for the kind of access to older data that is not indexed.